src/plugins/analysis/hardware_analysis/code/hardware_analysis.py

```diff
@@ -0,0 +1,115 @@
+                        elf_analysis = elf_analysis.replace("\x00", "\n")
+                        break
+                    
+                    # no .modinfo
+                    elf_analysis = "no information"
+
+            else:
+                # binary is nonetype
+                elf_analysis = "no information"
+
+        else:
+            elf_analysis = "no information"
+
+        # getting the kernel configs
+        kernel_config = kernel_config_dict.get("kernel_config")
+
+        if isinstance(kernel_config, str):
+
+            kernel_config_list = kernel_config.splitlines()
+
+            kernel_config = ""
+
+            for line in kernel_config_list:
+
+                if not line.startswith("#"):
+                    kernel_config = kernel_config + line + "\n"
+
+        else:
+            kernel_config = "no information"
+
+        # store the results
+        file_object.processed_analysis[self.NAME] = dict()
+        file_object.processed_analysis[self.NAME]['cpu architecture'] = cpu_architecture
+        file_object.processed_analysis[self.NAME]['*.ko ".modinfo"'] = elf_analysis
+        file_object.processed_analysis[self.NAME]['kernel configuration'] = kernel_config
+
+        # propagate some summary to parent objects
+
+        if cpu_architecture != "no information":
+
+            if (elf_analysis == "no information") & (kernel_config == "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = [f'{cpu_architecture}']
+
+            elif (elf_analysis != "no information") & (kernel_config != "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = [f'{cpu_architecture} - .modinfo available - kernel configuration available']
+
+            elif (elf_analysis != "no information") & (kernel_config == "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = [f'{cpu_architecture} - .modinfo available']
+
+            elif (elf_analysis == "no information") & (kernel_config != "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = [f'{cpu_architecture} - kernel configuration available']
+
+        else:
+            if (elf_analysis == "no information") & (kernel_config == "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = ['']
+
+            elif (elf_analysis != "no information") & (kernel_config != "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = ['.modinfo available - kernel configuration available']
+
+            elif (elf_analysis != "no information") & (kernel_config == "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = ['.modinfo available']
+
+            elif (elf_analysis == "no information") & (kernel_config != "no information"):
+                file_object.processed_analysis[self.NAME]['summary'] = ['kernel configuration available']
```

Comment:
The summary is a list of strings and we can simply add more entries where it makes sense. In this case it would probably be a good idea to add summary items for each category, i.e.
```suggestion
        if cpu_architecture != "no information":
            file_object.processed_analysis[self.NAME]['summary'].append(f'{cpu_architecture}')
        if elf_analysis != "no information":
            file_object.processed_analysis[self.NAME]['summary'].append('modinfo')
        if kernel_config != "no information":
            file_object.processed_analysis[self.NAME]['summary'].append('kernel configuration')
```

Reply:
On a side note: `&` is a binary logic operator and should only be used for binary logic operations. Instead you could simply use the "regular" logic operator  `and` (the same goes for `|` and `or`)